modules/dhp_connected_devices/spec/lib/fitbit_client_spec.rb

```diff
@@ -62,4 +81,20 @@
       expect(subject).to be_an_instance_of(described_class)
     end
   end
+
+  describe 'get_auth_code' do
+    it 'returns code as a string when auth code given' do
+      expect(subject.get_auth_code({ code: 'code' })).to eq('code')
+    end
+
+    it 'raise errors when auth code not given' do
+      error_params = { error: 'error', error_details: 'details' }
+
+      expect { subject.get_auth_code(error_params) }.to raise_error(DhpConnectedDevices::Fitbit::MissingAuthError)
+      expect { subject.get_auth_code({}) }.to raise_error(DhpConnectedDevices::Fitbit::MissingAuthError)
+      expect do
+        subject.get_auth_code({ random_param: '' })
+      end.to raise_error(DhpConnectedDevices::Fitbit::MissingAuthError)
```

Comment:
Question out of curiosity : is there a reason we us `do ... end` block instead of `{}` for the last case?

Reply:
This was a rubocop suggestion to work with line length. I've parameterized the error again, so the style can go back to the `{}`